### DISCUSSION TITLE
Prisma findmany by date

### DISCUSSION THREAD

**Discussion by abedaarabi** [prisma/prisma#13313] [Q&A] [answered]
Hi.
I am trying to load rows from the database using:
`const startDate = "4/20/2022"
const endDate = "4/27/2022"

const data = await prisma.dailyReport.findMany({
          where: {
            closingDate: {
              gte: String(startDate),
              lt: String(endDate),
            },
          },
        });`

the problem is that it restrives wired dates!
Am I using the right method in Prisma APIs?
![prisam](https://user-images.githubusercontent.com/45884020/168011158-f905d16b-632d-480c-98ac-dd13712ed053.PNG)

**Comment by richardwardza** [comment, marked as answer]
Is the `closingDate` column in your schema a `Date` or `String`?

From your example above it appears to be a `String` so your database is more than likely doing a string compare and not a date compare on the `gte` and `lt`.

Ideally `closingDate` would be a `Date` and then you can use:
```
const data = await prisma.dailyReport.findMany({
    where: {
      closingDate: {
        gte: new Date(startDate),
        lt: new Date(endDate),
      },
  },
});`
```

**Reply by abedaarabi** [reply]
it's a string, it's because I am interested in it from input (type date), I guess I agree with you it's doing a string compare!
but even though I am using `2022-05-12`  Prisma not accepting it !
thank you in advance

**Reply by lecheynata** [reply]
hi @abedaarabi it is correct it does not accept the data, since you would have to save them as datetime and applying the format utc time format to be able to use in a correct way the findMany, I do it this way using moment js:

Schema Model:
```mosel
id             Int       @id @default(autoincrement())
date           DateTime? @db.Date
missing        Boolean   @default(false)
createdAt      DateTime  @default(now())
updatedAt      DateTime? @updatedAt
```

Query:
```
const [ from, to ] = [ '2022-01-01', '2022-05-12' ]
const result = await prisma.assistance.findMany({
  where: {
    AND: [
      {
        date: {
          gte: moment.utc(from).format()
        }
      },
      {
        date: {
          lte: moment.utc(to).format()
        }
      }
    ]
  }
})
```

And remember that when saving you have to use the same utc format.

**Reply by abedaarabi** [reply]
now it make sense, thanks a lot

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
